File: lang/php/src/vac.fun/algorithms/number_theory/Factorial.php
<?php
    namespace vac\fun\algorithms\number_theory\Factorial;

    function iterativeProcedure(int $n): int {
        if ($n < 0) {
            return 0;
        }

        $result = 1;
        for ($i = 1; $i <= $n; $i++) {
            $result *= $i;
        }

        return $result;
    }

    function recursiveProcedure(int $n): int {
        if ($n < 0) {
            return 0;
        }

        if ($n == 0) {
            return 1;
        }
        return recursiveProcedure($n - 1) * $n;
    }
?>

File: go/muse/algorithms/number_theory/Factorial/Factorial.go
package Factorial

func IterativeProcedure(n int64) int64 {
	if n < 0 {
		return 0
	}

	result := int64(1)
	for i := int64(1); i <= n; i++ {
		result *= i
	}

	return result
}

func RecursiveProcedure(n int64) int64 {
	if n < 0 {
		return 0
	}

	if n == 0 {
		return 1
	}
	return RecursiveProcedure(n-1) * n
}
